mautic/mautic
# Added ignore tracked devices switch for shared computer page hit tracking.

| Q  | A
| --- | ---
| Bug fix? | No
| New feature? | Yes
| Automated tests included? | No
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | No
| Deprecations? | No

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Adds handling for an `ignore_tracked_devices` switch that can be passed in to the tracking JS `mt` function as a custom parameter. When its value is set to true Mautic will generate a new contact and tracking cookies for the device regardless of previous use. This switch when combined with the Mautic API 
can be used to track separate users on a shared computer provided the pages being tracked are able to distinguish different users for example via a user login. This is done by checking the `mtc_id` cookie value against the current user with the API to see if the cookie belongs to the them. If it does not belong `mt` is called with the `ignore_tracked_devices` switch generating a new contact. The new contact can then be assigned to the current user with the API.

#### Steps to test this PR:
1. Send multiple JS tracking requests to [this PR](https://mautibox.com) with the `ignore_tracked_devices` custom parameter set to true.
2. Verify the `mtc_id` cookie's value is updated each time.
